src/schema/schema_util.c

```diff
@@ -26,24 +26,24 @@ __wt_schema_backup_check(WT_SESSION_IMPL *session, const char *name)
 	conn = S2C(session);
 	if (!conn->hot_backup)
 		return (0);
-	__wt_readlock(session, &conn->hot_backup_lock);
-	/*
-	 * There is a window at the end of a backup where the list has been
-	 * cleared from the connection but the flag is still set.  It is safe
-	 * to drop at that point.
-	 */
-	if (!conn->hot_backup ||
-	    (backup_list = conn->hot_backup_list) == NULL) {
-		__wt_readunlock(session, &conn->hot_backup_lock);
-		return (0);
-	}
-	for (i = 0; backup_list[i] != NULL; ++i) {
-		if (strcmp(backup_list[i], name) == 0) {
-			ret = __wt_set_return(session, EBUSY);
+	WT_WITH_HOTBACKUP_LOCK(session, {
+		/*
+		 * There is a window at the end of a backup where the list has
+		 * been cleared from the connection but the flag is still set.
+		 * It is safe to drop at that point.
+		 */
+		if (!conn->hot_backup ||
+		    (backup_list = conn->hot_backup_list) == NULL) {
+			ret = 0;
 			break;
```

Comment:
Instead of explicitly unlocking and returning here, I'm setting `ret` and breaking which will unlock on the way out.